rdr_service/resource/generators/pdr_participant.py

```diff
@@ -0,0 +1,223 @@
+            for addr in res.addresses:
+                if addr['addr_type_id'] == StreetAddressTypeEnum.RESIDENCE.value:
+                    data['addr_city'] = addr['addr_city']
+                    data['addr_state'] = addr['addr_state']
+                    data['addr_zip'] = addr['addr_zip'][:3] if addr['addr_zip'] else addr['addr_zip']
+                    zipcode = addr['addr_zip']
+
+                    # See if we need to import the rural zip code list.
+                    if not self.rural_zipcodes:
+                        self._import_rural_zipcodes()
+                    if zipcode in self.rural_zipcodes:
+                        data['ubr_geography'] = 1
+
+        # ubr_education
+        if hasattr(res, 'education') and res.education:
+            if res.education in (
+                'HighestGrade_NeverAttended', 'HighestGrade_OneThroughFour', 'HighestGrade_NineThroughEleven',
+                'HighestGrade_FiveThroughEight'):
+                data['ubr_education'] = 1
+
+        # ubr_income
+        if hasattr(res, 'income') and res.income:
+            if res.income in ('AnnualIncome_less10k', 'AnnualIncome_10k25k'):
+                data['ubr_income'] = 1
+
+        # ubr_sexual_gender_minority
+        if data['ubr_sex'] == 1 or data['ubr_gender_identity'] == 1:
+            data['ubr_sexual_gender_minority'] = 1
+
+        # ubr_disability
+        qnans = ParticipantSummaryGenerator.get_module_answers(self.ro_dao, 'TheBasics', p_id)
+        data['ubr_disability'] = 0
+        if qnans:
+            if qnans.get('Employment_EmploymentStatus') == 'EmploymentStatus_UnableToWork' or \
+                qnans.get('Disability_Blind') == 'Blind_Yes' or \
+                qnans.get('Disability_WalkingClimbing') == 'WalkingClimbing_Yes' or \
+                qnans.get('Disability_DressingBathing') == 'DressingBathing_Yes' or \
+                qnans.get('Disability_ErrandsAlone') == 'ErrandsAlone_Yes' or \
+                qnans.get('Disability_Deaf') == 'Deaf_Yes' or \
+                qnans.get('Disability_DifficultyConcentrating') == 'DifficultyConcentrating_Yes':
+                data['ubr_disability'] = 1
+
+        # ubr_age_at_consent
+        if hasattr(res, 'date_of_birth') and res.date_of_birth and \
+            res.consents and len(res.consents) > 0:
+
+            consent_date = None
+            for consent_type in ['ConsentPII', 'EHRConsentPII_ConsentPermission', 'DVEHRSharing_AreYouInterested']:
+                if consent_date:
+                    break
+                for consent in res.consents:
+                    if consent['consent'] == consent_type and \
+                        consent['consent_value'] in ['ConsentPermission_Yes', 'DVEHRSharing_Yes']:
+                        consent_date = consent['consent_date']
+                        break
+
+            if consent_date:
+                age = int((consent_date - res.date_of_birth).days / 365)
```

Comment:
@pegbertsch made some recent updates for using float values when calculating the age (https://github.com/all-of-us/raw-data-repository/pull/2257/files#diff-f61c6da5bca97ae3faf169824db0c10a01f773e0a6082276cbe70bc525605f60R218) should that be used here as well? And should it potentially exist somewhere as a utility function they can share?

Should this consider leap years when determining age? I know it's been discussed before, but bringing it up again just in case it should be considered in a different light here.

Reply:
I missed this change, I've fixed up the resource generator with the same code.